examples/expect-extend/toBeWithinRange.ts

```diff
@@ -8,34 +8,39 @@
 import {expect} from '@jest/globals';
 import type {MatcherFunction} from 'expect';
 
-const toBeWithinRange: MatcherFunction<[floor: number, ceiling: number]> = (
-  actual: unknown,
-  floor: unknown,
-  ceiling: unknown,
-) => {
-  if (
-    typeof actual !== 'number' ||
-    typeof floor !== 'number' ||
-    typeof ceiling !== 'number'
-  ) {
-    throw new Error('These must be of type number!');
-  }
+const toBeWithinRange: MatcherFunction<[floor: number, ceiling: number]> =
+  function (actual: unknown, floor: unknown, ceiling: unknown) {
```

Comment:
ignore whitespace

Reply:
Yep I bumped into this limitation as well. The solution is not to use `this` or not to use arrow function:

https://github.com/facebook/jest/blob/ecab294cd5c1653041289b5cf80deea11af40f3f/packages/jest-snapshot/src/index.ts#L411-L412

Reply:
Alternatives for `this` could be: importing utils from modules and getting state from `global[STATE_SYMBOL]`? Or `expect.getState()`? Perhaps? Feels like lots of boilerplate code. `this` is cleaner solution.

Reply:
Yeh, I think assigning a function is fine. Same issue with React components and `React.FC` then `defaultProps` etc.  They don't have `this` to deal with, tho